packages/perseus/src/widgets/matrix/validate-matrix.ts

```diff
@@ -0,0 +1,30 @@
+import type {PerseusScore} from "../../types";
+import type {
+    PerseusMatrixUserInput,
+    PerseusMatrixValidationData,
+} from "../../validation.types";
+
+/**
+ * Checks user input from the matrix widget to see if it is scorable.
+ *
+ * Note: The matrix widget cannot do much validation without the Scoring
+ * Data because of its use of KhanAnswerTypes as a core part of scoring.
+ *
+ * @see `scoreMatrix()` for more details.
+ */
+function validateMatrix(
```

Comment:
Question: Did you decide just to leave all the `createValidator` logic in the scoring function? Do you think this is something we might want to change in the future? Looks like some of it is for scoring, but some is for validation.

Reply:
It's all left in the scoring function because there is no logic that can be moved into the validator that doesn't require the scoring data.

Reply:
It kind of looked like the `hasEmptyCell` check could possibly be moved out, but then you'd duplicate a lot of that.